### PR TITLE
feat: /x:description/attribute(result-file-threshold)

### LINKED ISSUE #398: Bug comparing expect dans result files that are the same
Hi,

attached a reproductible xspec that fails, and it shouldn't. [bug-xspec-impl-1.0.0_deep-equal.zip](https://github.com/xspec/xspec/files/2468127/bug-xspec-impl-1.0.0_deep-equal.zip)

Try running `xspec/aggregate_EE.xspec` ; test result presents on the left a file reference (dxx.xml) for the result and on the right part the expected content. It says result and expected are different, but I can't see where, as result is not displayed. I use XSpec distributed with Oxygen 20.1, it should be a very recent one, as there is support for schematron.

If you do not have suppressed intermediate files (you may have to comment out `<delete .../>`  in build.xml), you'll find generated files for result and expect contents.

Then, modify `reproductible.xsl`to load the two intermediates files. This xsl only call `test=deep-equal(...)` and presents the result. Run it, and check that the two files are identical.

So, I can not understand why the HTML report says result and expect are different.

Thanks in advance,
Christophe

### PR DESCRIPTION
**Pull request by AirQuick**
Closes #398
Closes #399

This pull request adds `@separate-results` to `/x:description`.

### Background

As of now, if an actual result or an expected result has 1000+ nodes, it is saved in a separate file. This behavior does not work very well:

- It's confusing.
  (See #398 and https://groups.google.com/g/xspec-users/c/dt8JTB01mBM/m/bGAlR63zAAAJ.)
- It is not compatible with the default **Run XSpec Test** transformation scenario.
  (See "_If you are running XSpec on Oxygen, the separate XML files may be deleted..._" in [Wiki](https://github.com/xspec/xspec/wiki/Understanding-Test-Results#result-xml).)

The behavior was introduced by e632690 whose commit message mentions two motivations:

- Ant has problems opening the report to check whether the tests have succeeded or not.
- It's also better for viewing the report, as these are now links to XML files rather than embedded serialised XML.

Probably those motivations were true when available resources were limited. I don't know if they still make sense today.

### Changes

So this pull request changes the behavior as follows:

- By default, the results are no longer saved as separate files.
- If you prefer the old behavior, you can set `/x:description/@separate-results`. If the result has more nodes than this attribute value, it is saved as a separate file.

### Further work
Update Wiki [`result-*.xml` section](https://github.com/xspec/xspec/wiki/Understanding-Test-Results#result-xml)

### CODE REVIEW

**Review by galtm — APPROVED** [review]
I like this change, and I agree that the existing behavior has several usability problems.

File: src/schemas/xspec.rnc [hunk 1]
```diff
@@ -43,6 +43,15 @@ description =
 		[ a:defaultValue = "no" ]
 		attribute measure-time { boolean.datatype }?,
 		
+		## This attribute determines whether to save results as separate files or not.
+		## If the number of nodes in an actual result or an expected result (after wrapped in a
```
[the remark this answers is not in the document]
Comment (AirQuick) [reply src/schemas/xspec.rnc]: Updated in ef2a508b0fa3a382f281c913ec05995f1267ca6d.

File: src/schemas/xspec.rnc [hunk 2]
```diff
@@ -43,6 +43,15 @@ description =
...
+		## report element) approximately exceeds this attribute, the result is saved as a separate
+		## file.
+		## Results are never saved as separate files, if the attribute value is 'never' or when
+		## testing XQuery.
+		[ a:defaultValue = "never" ]
+		attribute separate-results { "never" | xsd:positiveInteger }?,
```
[the remark this answers is not in the document]
Comment (AirQuick) [reply src/schemas/xspec.rnc]: Thanks for the ideas. That's what I was looking for.
88c2bdd2874faa3fd8f36e28e7f9e31474ab5ed2 takes `result-file-threshold` and `inf`.

> I don't even know how often users will use this attribute with a nondefault value; personally, I like the "never" behavior.

Agreed. This feature will be only for rare situations.

**Comment by AirQuick** [comment]
> ### Further work
> Update Wiki [`result-*.xml` section](https://github.com/xspec/xspec/wiki/Understanding-Test-Results#result-xml)

[Done](https://github.com/xspec/xspec/wiki/Understanding-Test-Results/_compare/6ae4bacf55f6db8b36ee7d5a01211bf6c6b507c7...57fcd8df57fdbdc01a391fd843a36aa82d85c1c8).

I think people will be only interested in the default behavior. So I just labelled the section as "only on XSpec v2.2.0 or earlier".

**Comment by galtm** [comment]
Interesting that the new attribute will be undocumented (except Oxygen tooltips). I agree with your judgment call, though. If surprisingly many users ask how to restore the old behavior, the wiki can always be changed.

**Pull request merged**